### ISSUE TITLE
Duplicate markers in clusters when zoomed in

### ISSUE THREAD

**Issue by JoshNewham1** [yuzhva/react-leaflet-markercluster#108]
_I'm not sure whether this problem is to do with the React wrapper or the main MarkerCluster library, just putting it on here because it seems like it could be a problem with adding dynamic JSX elements..._

I have a React app which uses this plugin with dynamic JSX to group circles and markers into clusters and improve the performance of the map. I have quite a large dataset (1078 markers and circles in total) and the clusters/groups are of the proper size when zoomed out to around zoom level 10. The problem lies when I zoom into an area and the groups recalculate and split out. It seems that duplicate markers are introduced at this point as the spider view of most clusters shows multiple duplicate markers and circles. Then, when zoomed in further to the point that the clusters disappear, the correct placement and number of circles/markers is restored.

Here is some [demo ](https://codesandbox.io/s/react-leaflet-5yo3t?file=/src/index.js) code showing the problem with a large dataset. It doesn't seem to happen as much where the clusters are smaller, but in London and around Hyde Park there are plenty of duplicate markers and circles.

I can't see it being a problem with how I load and use my dataset but rather the size and density of the points.

Any help or suggestions would be appreciated,

Thanks for your time, Josh

_Example images of problem:_

![1](https://user-images.githubusercontent.com/25721451/79682529-faff0900-821a-11ea-818f-f285948fe396.JPG)  ![2](https://user-images.githubusercontent.com/25721451/79682528-fa667280-821a-11ea-93d1-39f4c1dff19e.JPG)

![3](https://user-images.githubusercontent.com/25721451/79682530-fb979f80-821a-11ea-9f4a-c2ff16ca6541.JPG)  ![4](https://user-images.githubusercontent.com/25721451/79682531-fb979f80-821a-11ea-9951-105ba53fda6d.JPG)

**Comment by yuzhva** [comment]
<img width="939" alt="Screenshot 2020-04-20 at 09 45 12" src="https://user-images.githubusercontent.com/5191620/79721834-b2fbe700-82eb-11ea-8c4e-504d39ea1a7d.png">

The number of elements from state arrays:
446+627=1073

Are equal to the number of Clustered markers
1073

> It seems that duplicate markers are introduced at this point as the spider view of most clusters shows multiple duplicate markers and circles. 
The markers are not duplicated.

Just they are placed too close to each other, - the coordinates are very close.
_Try to update formula for markers generation._

**Comment by JoshNewham1** [comment]
Yeah, it doesn't seem as if the marker/circle objects themselves are duplicated in the array, rather that the clustering algorithm doesn't work correctly with the current density of the points. I've updated the radius filter to what you have suggested and it hasn't really made a difference. Unfortunately, I can't really reduce the density of my points unless I added some sort of user filter which wouldn't be ideal. Since opening this issue, I experimented further and found that this problem occurs with vanilla JavaScript as well as in React so it seems like it's a problem with the clustering algorithm from the Leaflet.MarkerCluster library.

Thanks for the suggestion though,
Josh
